webapp/app/forms/steps/lotse/personal_data.py

```diff
@@ -166,3 +171,125 @@ def render(self, **kwargs):
                                # TODO: These are still required by base.html to set the page title.
                                form=self.render_info.form,
                                header_title=self.header_title)
+
+
+class ShowPersonBPrecondition(FamilienstandModel):
+    _step_to_redirect_to = StepFamilienstand.name
+    _message_to_flash = _l('form.lotse.skip_reason.familienstand_single')
+
+    @root_validator(skip_on_failure=True)
+    def person_b_must_be_shown(cls, values):
+        if not JointTaxesModel.show_person_b(values):
+            raise ValidationError
+        return values
+
+
+class StepPersonB(LotseFormSteuerlotseStep):
+    name = 'person_b'
+    title = _l('form.lotse.person-b-title')
+    intro = _l('form.lotse.person-b-intro')
+    header_title = _l('form.lotse.mandatory_data.header-title')
+    # TODO remove this once the adjacent steps are converted to steuerlotse steps
+    prev_step = StepPersonA
+    next_step = StepIban
+
+    label = _l('form.lotse.step_person_b.label')
+    section_link = SectionLink('mandatory_data', StepFamilienstand.name, _l('form.lotse.mandatory_data.label'))
+
+    template = 'lotse/form_person_b.html'
+
+    preconditions = [ShowPersonBPrecondition]
+
+    class InputForm(SteuerlotseBaseForm):
+        def input_required_if_not_same_address(form, field):
+            if form.person_b_same_address.data == 'yes':
+                validators.Optional()(form, field)
+            else:
+                validators.InputRequired()(form, field)
+
+        def validate_person_b_beh_grad(self, field):
+            if self.person_b_gehbeh.data:
+                validators.InputRequired(_l('form.lotse.validation-person-beh-grad'))(self, field)
+            else:
+                validators.Optional()(self, field)
```

Comment:
I know you did not change this. However, shouldn't we put these below the declaration of the fields?